opal/threads/wait_sync.h

```diff
@@ -27,50 +29,44 @@ typedef struct ompi_wait_sync_t {
 #define REQUEST_COMPLETED      (void*)1L
 
-#if OPAL_ENABLE_MULTI_THREADS
-
-#define OPAL_ATOMIC_ADD_32(a,b)         opal_atomic_add_32(a,b)
-#define OPAL_ATOMIC_SWP_PTR(a,b)        opal_atomic_swap_ptr(a,b)
-#define SYNC_WAIT(sync)                 sync_wait_mt(sync)
-#define PTHREAD_COND_INIT(a,b)          pthread_cond_init(a,b)
-#define PTHREAD_MUTEX_INIT(a,b)         pthread_mutex_init(a,b)
+#define SYNC_WAIT(sync)                 (opal_using_threads() ? sync_wait_mt (sync) : sync_wait_st (sync))
+#define PTHREAD_COND_INIT(a,b)          (opal_using_threads() ? pthread_cond_init (a,b) : 0)
+#define PTHREAD_MUTEX_INIT(a,b)         (opal_using_threads() ? pthread_mutex_init (a,b) : 0)
 
 #define WAIT_SYNC_RELEASE(sync)                       \
-    do {                                              \
+    if (opal_using_threads()) {                       \
        pthread_cond_destroy(&(sync)->condition);      \
        pthread_mutex_destroy(&(sync)->lock);          \
-    } while(0)
+    }
 
 #define WAIT_SYNC_SIGNAL(sync)                        \
-    do {                                              \
+    if (opal_using_threads()) {                       \
         pthread_mutex_lock(&(sync->lock));            \
         pthread_cond_signal(&sync->condition);        \
         pthread_mutex_unlock(&(sync->lock));          \
-    } while(0)
-
-#else
+    }
 
-#define OPAL_ATOMIC_ADD_32(a,b)         (*(a) += (b))
-#define OPAL_ATOMIC_SWP_PTR(a,b)        *(a) = (b)
-#define PTHREAD_COND_INIT(a,b)
-#define PTHREAD_MUTEX_INIT(a,b)
-#define SYNC_WAIT(sync)                 sync_wait_st(sync)
-#define WAIT_SYNC_RELEASE(sync)
-#define WAIT_SYNC_SIGNAL(sync)
+OPAL_DECLSPEC int sync_wait_mt(ompi_wait_sync_t *sync);
+static inline int sync_wait_st (ompi_wait_sync_t *sync)
+{
+    while (sync->count > 0) {
+        opal_progress();
+    }
 
-#endif /* OPAL_ENABLE_MULTI_THREADS */
+    return sync->status;
+}
 
-OPAL_DECLSPEC int sync_wait_mt(ompi_wait_sync_t *sync);
-OPAL_DECLSPEC int sync_wait_st(ompi_wait_sync_t *sync);
 
-#define WAIT_SYNC_INIT(sync,c)                        \
-    do {                                              \
-       (sync)->count = c;                             \
-       (sync)->next = NULL;                           \
-       (sync)->prev = NULL;                           \
-       (sync)->status = 0;                            \
-       PTHREAD_COND_INIT(&(sync)->condition, NULL);   \
-       PTHREAD_MUTEX_INIT(&(sync)->lock, NULL);       \
+#define WAIT_SYNC_INIT(sync,c)                                  \
+    do {                                                        \
+        (sync)->count = c;                                      \
+        (sync)->next = NULL;                                    \
+        (sync)->prev = NULL;                                    \
+        (sync)->status = 0;                                     \
+        if (opal_using_threads()) {                             \
```

Comment:
Indeed. I didn't think to check what those macro did. Will clean it up in a bit.